### PR TITLE
Invite user api

### PR DESCRIPTION
**Pull request by shalapatil**
## Summary
https://www.notion.so/saeloun/Develop-API-to-add-new-team-member-69a4e28d55ea4f39918d6b6e124c16dd

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to
      not work as expected)
- [ ] This change requires a documentation update

### Checklist:

- [ ] I have manually tested all workflows
- [x] I have performed a self-review of my own code
- [x] I have added automated tests for my code

### CODE REVIEW

**Review by rohitjoshixyz — APPROVED** [review]
LGTM 🎉

File: app/controllers/internal_api/v1/users_controller.rb [hunk 1]
```diff
@@ -1,28 +1,37 @@
...
+  include ErrorHandler
+  include CurrentCompanyConcern
+
+  before_action :authenticate_user!
   after_action :assign_company, only: [:create]
   after_action :assign_role, only: [:create]
 
+  def create
+    authorize User
+    super do |resource|
```
Comment (akhilgkrishnan) [inline app/controllers/internal_api/v1/users_controller.rb:14]: @shalapatil can you please confirm whether the invitation emails are sending or not?

Comment (shalapatil) [reply app/controllers/internal_api/v1/users_controller.rb:14]: @akhilgkrishnan Yes, emails are being sent!

**Comment by supriya3105** [comment]
@akhilgkrishnan @rohitjoshixyz Please review and approve if no further review comments.

**Review by akhilgkrishnan — APPROVED** [review]

**Comment by akhilgkrishnan** [comment]
> @akhilgkrishnan @rohitjoshixyz Please review and approve if no further review comments.

@supriya3105 Do not merge this PR without testing the entire user invite flow.

**Comment by supriya3105** [comment]
@akhilgkrishnan We need to merge this on the feature branch created by @ajinkyaa for the teams page refactoring. @shalapatil @ajinkyaa This PR needs to be merged into feature branch. Please coordinate among yourselves. Do not merge this on develop branch

**Review by rohitjoshixyz — APPROVED** [review]
LGTM

File: app/controllers/internal_api/v1/users_controller.rb [hunk 2]
```diff
@@ -1,28 +1,37 @@
...
+        render json: resource.errors, status: 401 and return
+      end
+    end
+  end
+
   protected
 
     def assign_company
-      unless invited_user.errors.present? ||
-          invited_user.companies.exists?(id: current_company.id)
```
Comment (apoorv-mishra) [inline app/controllers/internal_api/v1/users_controller.rb:11]: Why were these checks removed @shalapatil ? Can you test the flow where a user is invited to same workspace multiple times @apoorv1316 for this PR? It shouldn't happen.

**Review by apoorv-mishra — CHANGES_REQUESTED** [review]

File: app/controllers/internal_api/v1/users_controller.rb (hunk 2 above)
Comment (apoorv1316) [reply app/controllers/internal_api/v1/users_controller.rb:11]: @apoorv-mishra I invited a user as an employee and without accepting the invitation I invited him again in the same workspace and it sent the invitation again. cc @shalapatil @supriya3105 @akhilgkrishnan

**Pull request closed without merging**